app/controllers/pages_controller.rb

```diff
@@ -44,21 +43,17 @@ def analytics
 
   def get_data
     begin
-      @property = @service.get_web_property(params[:account_id],
-                                            params[:web_property_id])
-
-      @image = ReportGenerator.generate(@service, params, @property)
-      message = "#{session[:user_name]} generated a report for #{@property.name}"
+      asset = Asset.new(@service, params)
+      @property = asset.property
+      @image = asset.image
       render :analytics
     rescue Google::Apis::AuthorizationError => ex
       if ex.message
-        render jason: {message: ex}
```

Comment:
on a more serious note, behaviour has changed without a change in tests, this is a good indication that certain tests are missing 😉